### DISCUSSION TITLE
Version 7.34.0-next.0

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#8659] [General]
🎪 feature request: #6879 `useFieldArray` support `rules` props (#8102)

```tsx
useFieldArray({
  name: 'test',
  rules: {
    required: true,
    minLength: 2,
    maxLength: 10,
    validate: (fieldArrayValues) => {
      if (fieldArrayValues[2].title === 'test') {
        return 'validate Error'
      }
    }
  }
})

errors?.test?.root?.message // access root level errors
```

**Note:** Related context and poll for error message naming: https://github.com/react-hook-form/react-hook-form/discussions/8625

<hr /><em>This discussion was created from the release <a href='https://github.com/react-hook-form/react-hook-form/releases/tag/v7.34.0-next.0'>Version 7.34.0-next.0</a>.</em>

**Comment by Moshyfawn** [comment]
Does this feature include the type-safty for the root field array error type? Like, if I validate the length of the field array..
Will take a look at the code on Monday; maybe someone can just onboard me in regards to the question above :P Thank you :D

**Reply by bluebill1049** [reply]
It should, try it out when you got a chance. :)
